packages/rum-core/src/domain/rumEventsCollection/action/trackClickActions.ts

```diff
@@ -95,79 +105,97 @@ export function trackClickActions(
     actionContexts,
   }
 
+  function appendClickToClickChain(click: Click) {
+    if (!currentClickChain || !currentClickChain.tryAppend(click)) {
+      const rageClick = click.clone()
+      currentClickChain = createClickChain(click, (clicks) => {
+        finalizeClicks(clicks, rageClick)
+      })
+    }
+  }
+
   function stopClickChain() {
     if (currentClickChain) {
       currentClickChain.stop()
     }
   }
+}
 
-  function processClick({ event, getUserActivity }: OnClickContext) {
-    if (!trackFrustrations && history.find()) {
-      // TODO: remove this in a future major version. To keep retrocompatibility, ignore any new
-      // action if another one is already occurring.
-      return
-    }
+function onPointerDown(
+  configuration: RumConfiguration,
+  lifeCycle: LifeCycle,
+  domMutationObservable: Observable<void>,
+  history: ClickActionIdHistory,
+  stopObservable: Observable<void>,
+  appendClickToClickChain: (click: Click) => void,
+  pointerDownEvent: MouseEventOnElement
+) {
+  if (!configuration.trackFrustrations && history.find()) {
+    // TODO: remove this in a future major version. To keep retrocompatibility, ignore any new
+    // action if another one is already occurring.
+    return
+  }
 
-    const clickActionBase = computeClickActionBase(event, configuration.actionNameAttribute)
-    if (!trackFrustrations && !clickActionBase.name) {
-      // TODO: remove this in a future major version. To keep retrocompatibility, ignore any action
-      // with a blank name
-      return
-    }
+  const clickActionBase = computeClickActionBase(pointerDownEvent, configuration.actionNameAttribute)
+  if (!configuration.trackFrustrations && !clickActionBase.name) {
+    // TODO: remove this in a future major version. To keep retrocompatibility, ignore any action
+    // with a blank name
+    return
+  }
 
-    const click = newClick(lifeCycle, history, getUserActivity, clickActionBase)
+  return {
+    onClick({ event: clickEvent, getUserActivity }: OnClickContext) {
```

Comment:
> **💭 thought:** ‏IMO the current API is a bit hard to follow especially the optional onClick to indicate that a click has been aborted.
> 
> Could we keep all the processing at click level but provides both pointerdown and click events to the processing? listenActionEvents would only be responsible to keep this event and provide it later.

No, as we want to compute the CSS selector before the UI reacts to the click (second motivation in the PR description 🙂 )

> 
> Otherwise, what about splitting the processing of the pointer down from the processing of the click? We could rely on the result of the former to consider the click as aborted or not.

I'll make something.